### PR TITLE
Add link to grammar.txt in the documentation

### LINKED ISSUE #1176: Include grammar.txt in the documentation
This issue is a spin-off of @chriseth's [comment](https://github.com/ethereum/solidity/issues/315#issuecomment-242554903).  The proposal is to include grammar.txt in the documentation and to list the reserved keywords in the same section.

### PR DESCRIPTION
**Pull request by axic**
Fixes #1176.

### CODE REVIEW

File: docs/miscellaneous.rst [hunk 1]
```diff
@@ -347,3 +347,8 @@ These keywords are reserved in Solidity. They might become part of the syntax in
 
 ``abstract``, ``after``, ``case``, ``catch``, ``final``, ``in``, ``inline``, ``interface``, ``let``, ``match``,
 ``of``, ``pure``, ``relocatable``, ``static``, ``switch``, ``try``, ``type``, ``typeof``, ``view``.
+
+Language Grammar
+================
+
+The entire language grammar is `available here <https://github.com/ethereum/solidity/blob/develop/libsolidity/grammar.txt>`_.
```
Comment (axic) [inline docs/miscellaneous.rst]: Perhaps it should be a link to the `release` branch and not `develop`?

**Comment by axic** [comment]
Should we also include a link to the unofficial peg.js: https://github.com/ConsenSys/solidity-parser/blob/master/solidity.pegjs?

**Comment by chriseth** [comment]
I think we should move grammar.txt into the documentation: https://docs.python.org/3/reference/grammar.html

**Comment by axic** [comment]
@chriseth the python one to me seems (based on the comments in the file) to be kept as a regular file and copied into the documentation through a build script.

**Comment by chriseth** [comment]
@axic the python file is used by the parser, but I actually don't see a single reason why `grammar.txt` should be its own file for solidity.

**Comment by axic** [comment]
As long as it can be placed in a `<pre>` which still allows it to be copy pasted without changes I'm fine. However I do not see a problem having it as a link only as in this PR.

**Comment by chriseth** [comment]
`grammar.txt` is the only part of the documentation that is not part of the documentation.

**Comment by pirapira** [comment]
The move of `grammar.txt` is tracked in #1206.

**Comment by pirapira** [comment]
The addition of the link is currently a good change.

**Comment by HaleTom** [comment]
`grammar.txt` is now at: https://github.com/ethereum/solidity/blob/develop/docs/grammar.txt

**Comment by axic** [comment]
The grammar is actually inlined in the documentation.

**Pull request merged**